Sources/SwiftGraphQLParser/GraphQLParser.swift

```diff
@@ -30,17 +30,79 @@ public struct UnexpectedTokenError: LocalizedError {
 
 public struct ParserError: LocalizedError {
 	public let type: ParserErrorType
-	public let line: Int
+	public let errorRange: Range<String.Index>
+	public let input: String
+	public var line: Int
 	public let column: Int
 	
+	let contextualStart: String.Index
```

Comment:
what are the differences between the errorRange, and the contextualStart/end indices?

Reply:
`errorRange` is where the parser fails and `contextualStart` and `contextualEnd` gives a larger context for where the parser was when parsing.  For example, in `testEmptyArgumentList`, this fragment

```graphql
fragment Fragment on Customer {
    customers() {}
}
```

yields `errorRange` starting on the open parenthesis and ending on the closed parenthesis while the contextual start and end encompasses the following block:

```graphql
fragment Fragment on Customer {
    customers()
```

Reply:
I will add contextual checks in the tests to make this difference more apparent.